### PR TITLE
NMS 14671 Add adoc documentation for partial configuration update

### PR DESCRIPTION
**Pull request by dherdt**
* JIRA (Issue Tracker): https://issues.opennms.org/browse/NMS-14671

### CODE REVIEW

File: docs/modules/development/pages/rest/config_management.adoc [hunk 1]
```diff
@@ -38,6 +46,16 @@ Configurations modified will take effect instantly.
 200: Return nothing
 
 400: Return: {"MESSAGE": "<ERROR_MESSAGE>"}
+
+| `/cm/\{configName}/\{configId}/\{path}` |
+Adds an array element to the configuration part specified by configName, configId and the path to the part in JsonPath format. Returns 404 when the configuration does not exist. Returns 400 When the path parameter has a wrong format or specifies a nonexistent path or the provided part of configuration has a wrong format.
```
Comment (Bonrob2) [inline docs/modules/development/pages/rest/config_management.adoc:51]:
```suggestion
Adds an array element to the configuration part specified by `configName`, `configId`, and the path to the part in JSONPath format.
Returns 404 when the configuration does not exist.
Returns 400 when the path parameter has a wrong format, specifies a nonexistent path, or the provided part of the configuration has a wrong format.
```

Same questions as above regarding use of the word "part".

File: docs/modules/development/pages/rest/config_management.adoc [hunk 2]
```diff
@@ -48,9 +66,27 @@ Configurations modified will take effect instantly.
 | `/cm/\{configName}/\{configId}`             |  Update configuration.
                                                      If configuration does not exist, it will return error.
 
-200: Return nothing
+200: Return: nothing
 
 400: Return: {"MESSAGE": "<ERROR_MESSAGE>"}
+| `/cm/\{configName}/\{configId}/\{path}` |
+Updates a part of configuration specified by configName, configId and the path to the part in JsonPath format. Returns 404 when the configuration does not exist. Returns 400 When the path parameter has a wrong format or specifies a nonexistent path or provided part of configuration has a wrong format.
```
Comment (Bonrob2) [inline docs/modules/development/pages/rest/config_management.adoc:73]:
```suggestion
Updates a part of the configuration specified by `configName`, `configId`, and the path to the part in JSONPath format.
Returns 404 when the configuration does not exist.
Returns 400 when the path parameter has a wrong format, specifies a nonexistent path, or the provided part of the configuration has a wrong format.
```
Same question about the word "path".

File: docs/modules/development/pages/rest/config_management.adoc [hunk 3]
```diff
@@ -48,9 +66,27 @@ Configurations modified will take effect instantly.
...
+
+200 Return: nothing
+
+404 Return: {"MESSAGE": "<ERROR_MESSAGE>"}
+
+400 Return: {"MESSAGE": "<ERROR_MESSAGE>"}
+
+| `/cm/\{configName}/\{configId}/\{pathToParent}/\{nodeName}` |
+Updates or inserts a part of configuration specified by configName, configId, path to the parent node in JsonPath format and node name. Returns 404 when the configuration does not exist. Returns 400 When the pathToParent parameter has a wrong format or specifies a nonexistent path or provided part of configuration has a wrong format.
```
Comment (Bonrob2) [inline docs/modules/development/pages/rest/config_management.adoc:82]:
```suggestion
Updates or inserts a part of the configuration specified by `configName`, `configId`, and the path to the parent node in JSONPath format with the node name.
Returns 404 when the configuration does not exist.
Returns 400 when the `pathToParent` parameter has a wrong format, specifies a nonexistent path, or the provided part of the configuration has a wrong format.
```
In the first sentence, is the path to the parent node in JSONPath format AND includes the node name? If so, please use my suggestion of "with".

Earlier examples do not specify the `pathToParent` parameter. Is this intentional?

File: docs/modules/development/pages/rest/config_management.adoc [hunk 4]
```diff
@@ -63,4 +99,12 @@ Configurations modified will take effect instantly.
 200: Return nothing
 
 400: Return: {"MESSAGE": "<ERROR_MESSAGE>"}
+| `/cm/\{configName}/\{configId}/\{path}` |
+Deletes a part of the configuration specified by configName, configId and the path to the part in JsonPath format. Returns 404 when the configuration does not exist. Returns 400 When the path parameter has a wrong format or specifies a nonexistent path or the provided part of configuration has a wrong format.
```
Comment (Bonrob2) [inline docs/modules/development/pages/rest/config_management.adoc:103]:
```suggestion
Deletes a part of the configuration specified by `configName`, `configId`, and the path to the part in JSONPath format.
Returns 404 when the configuration does not exist.
Returns 400 when the path parameter has a wrong format, specifies a nonexistent path, or the provided part of the configuration has a wrong format.
```
Same questions as before about the word "part".

**Review by Bonrob2 — CHANGES_REQUESTED** [review]
A few changes. Thanks!

File: docs/modules/development/pages/rest/config_management.adoc [hunk 5]
```diff
@@ -24,6 +24,14 @@ Configurations modified will take effect instantly.
 200: Return: Config Bean 
 
 400: Return: {"MESSAGE": "<ERROR_MESSAGE>"}
+| `/cm/\{configName}/\{configId}/\{path}` |
+Returns a part of configuration specified by configName, configId and the path to the part in JsonPath format. Returns 404 when the configuration does not exist. Returns 400 When the path parameter has a wrong format or specifies a nonexistent path.
```
[the remark this answers is not in the document]
Comment (dherdt) [reply docs/modules/development/pages/rest/config_management.adoc]: There are two Versions of PUT: with whole "path" and consisting of "pathToParent" and "nodeName".
1: "$.path.to.somting"
2: "$.path.to" and "somting"
the second version works also when "somting" do not exist in note "to"

**Review by Bonrob2 — COMMENTED** [review]
I'm still not 100% clear on what the "part" of the configuration indicates and how that "part" is determined. I am guessing that users will understand?

**Review by Bonrob2 — APPROVED** [review]

**Pull request merged**